drivers/azure/azureutil/auth.go

```diff
@@ -33,11 +29,25 @@ var (
 // will automatically refresh the specified token and will call the refresh
 // callback function we implement here. This way we will always be storing a
 // token with a refresh_token saved on the machine.
+//
+// 2. Azure Service Principal Account
+//
+// This is designed for headless authentication to Azure APIs but requires more
+// steps from user to create a Service Principal Account and provide its
+// credentials to the machine driver.
+
+var (
+	// AD app id for docker-machine driver in various Azure realms
+	appIDs = map[string]string{
+		azure.PublicCloud.Name: "637ddaba-219b-43b8-bf19-8cea500cf273",
```

Comment:
@ppadala just so you know this piece of code is actually not modified at all, I just renamed the variable.

But some explanation: these IDs ares only for browser-based authentication. In that case, we have a Docker-Machine application created in ActiveDirectory. Obviously, databases are separate for each Azure realm (e.g. Azure Public Cloud, Azure China, Azure Germany, Azure US Government etc). So we just create AAD apps when people ask for (like it happened in the China case).

If you use Service Principal Account, these app IDs are not used.